src/floor/FloorSystem.java

```diff
@@ -0,0 +1,146 @@
+package floor;
+
+import java.io.IOException;
+import java.net.DatagramPacket;
+import java.net.DatagramSocket;
+import java.net.SocketException;
+
+import floor.Floor.directionLampState;
+import messages.ElevatorRequestMessage;
+import messages.ElevatorRequestMessage.Direction;
+import messages.FloorArrivalMessage;
+import messages.FloorTravelMessage;
+import messages.Message;
+
+public class FloorSystem {
```

Comment:
This class feels a bit odd to me because we're supposed to have a one-floor subsystem per floor, but this class is modeling the whole thing. Do you have a reason for this decision?